include/z64bgcheck.h

```diff
@@ -0,0 +1,203 @@
+#ifndef _Z_BGCHECK_
+#define _Z_BGCHECK_
+
+struct GlobalContext;
+struct Actor;
+struct DynaPolyActor;
+
+#define SS_NULL 0xFFFF
+#define COLPOLY_NORM_FRAC (1.0f / 32767)
+#define COLPOLY_GET_NORM(n) ((n)*COLPOLY_NORM_FRAC)
+
+// bccFlags
+#define BGCHECK_CHECK_WALL 1
+#define BGCHECK_CHECK_FLOOR 2
+#define BGCHECK_CHECK_CEILING 4
+#define BGCHECK_CHECK_ONE_FACE 8
+#define BGCHECK_CHECK_DYNA 0x10
+
+// bciFlags
+#define BGCHECK_IGNORE_NONE 0
+#define BGCHECK_IGNORE_CEILING 1
+#define BGCHECK_IGNORE_WALL 2
+#define BGCHECK_IGNORE_FLOOR 4
+
+// xpFlags
+#define COLPOLY_IGNORE_NONE 0
+#define COLPOLY_IGNORE_CAMERA 1
+#define COLPOLY_IGNORE_ENTITY 2
+#define COLPOLY_IGNORE_PROJECTILES 4
+#define COLPOLY_SNORM(x) ((short)((x) * 32767))
+
+#define BG_ACTOR_MAX 50
+#define BGCHECK_SCENE BG_ACTOR_MAX
```

Comment:
BGCHECK_SCENE looks pretty confusing to me when i read it in the code, what it meant by this name?
BG_ACTOR_MAX makes way more sense to me for what it does

Reply:
BGCHECK_SCENE is meant for code that is intending to access the scene collision data; using BG_ACTOR_MAX wouldn't be appropriate in those cases.

Reply:
ok so scene collision data comes right after dyna stuff? thats good to know. I feel like we might be able to explain that in a comment so that people using it are aware that we need to pick the right one depending on the context